src/coreclr/jit/ssabuilder.cpp

```diff
@@ -115,10 +115,24 @@ void Compiler::fgResetForSsa()
             {
                 if (tree->IsLocal() || tree->OperIsLocalAddr())
                 {
+                    GenTreeLclVarCommon* lclNode = tree->AsLclVarCommon();
+                    unsigned lclNum = lclNode->GetLclNum();
+                    LclVarDsc* varDsc = lvaGetDesc(lclNum);
+
+                    if (varDsc->lvPromoted)
+                    {
+                        for (unsigned index = 0; index < varDsc->lvFieldCnt; index++)
+                        {
+                            unsigned   fieldLclNum = varDsc->lvFieldLclStart + index;
+                            if (!lvaInSsa(fieldLclNum))
+                            {
+                                lclNode->SetSsaNum(this, index, SsaConfig::RESERVED_SSA_NUM);
+                            }
+                        }
+                    }
```

Comment:
Hmm, it is very surprising this is needed. `SetSsaNum(SsaConfig::RESERVED_SSA_NUM);` should be enough to clear out the SSA number, composite or not.